syntax = "proto3";

import "google/protobuf/wrappers.proto";

package proto;

service TicketInfo {
    rpc createTicket(Ticket) returns (TicketID);
    rpc getTicket(TicketID) returns (Ticket);
    rpc getTickets(TicketRange) returns (stream Ticket);
	rpc updateTicket(stream Ticket) returns (google.protobuf.StringValue);
	rpc getTicketOneByOne(stream google.protobuf.StringValue) returns (stream Ticket);
}

message Ticket {
    int64 id = 1;
    string creator = 2;
    string note = 3;
    string variable = 4;
}

message TicketID {
    int64 id = 1;
}

message TicketRange {
    int64 low = 1;
    int64 high = 2;
}
